### ISSUE TITLE
[Feature Request] Coasting (Reopen Feature Request #57)

### ISSUE THREAD

**Issue by robschwieb** [supermerill/SuperSlicer#517]
#57 

The above feature request is closed and marked as "fix is live in the last release" but I don't think there was a good understanding of the original request as explained by VanessaE [here.](https://github.com/supermerill/SuperSlicer/issues/57#issuecomment-684674438)

This is something that would add a lot to SuperSlicer over PrusaSlicer considering S3D, Cura, and IdeaMaker have had it for a long time. Prusa people have been requesting it for over three years [here.](https://github.com/prusa3d/PrusaSlicer/issues/246)

**Comment by supermerill** [comment]
ok,ok, i'll add a slicer-based pressure model.

**Comment by robschwieb** [comment]
I don't think coasting is considered a "pressure model".

I've cranked up the coasting distance to 3mm in IdeaMaker for easy visual representation but hopefully this clear it up a bit. You can see how the last 3mm of the extrusion is canceled/removed.

![image](https://user-images.githubusercontent.com/57918897/94847855-23597900-03f1-11eb-91b0-ae5b80da6c8a.png)

**Comment by robschwieb** [comment]
It's not nearly as complex as that. At least not in the slicers I've used it with. Just a user specified distance (usually very small like 0.4mm) or specified volume like in Cura which recommends your nozzle diameter cubed as the value.

![image](https://user-images.githubusercontent.com/57918897/94848416-f6f22c80-03f1-11eb-8102-01513a215b45.png)

**Comment by SnowDrifterr** [comment]
I think coasting by volume with a commanded coast speed might be most appropriate. 

I'm not sure how fruitful it would be to try to implement a pressure model given that pressure should be more or less corrected by linear advance / pressure advance. This just deals with nozzle ooze and aiding to reduce the appearance of Z seams.

With that said, I feel that a coasting calibration with this method should be relatively simple to set up. It would be as simple as printing a cube with varying coast amounts that increases by Z height, using an on-face seam alignment for reference (e.g. "back"). Or slice up a couple objects and print them individually, similar to how extrusion calibration is performed.

**Comment by supermerill** [comment]
> I'm not sure how fruitful it would be to try to implement a pressure model given that pressure should be more or less corrected by linear advance / pressure advance. 

If the pressure is corrected by linear advance, then there is no need for coasting.

**Comment by robschwieb** [comment]
> > I'm not sure how fruitful it would be to try to implement a pressure model given that pressure should be more or less corrected by linear advance / pressure advance.
> 
> If the pressure is corrected by linear advance, then there is no need for coasting.
> Ooze is too slow and unreliable to create the end of the extrusion.

Even with a perfectly dialed linear/pressure advance coasting still comes in handy. I get my best looking seams in S3D using just a tad of coast on top of pressure advance.

**Comment by SnowDrifterr** [comment]
> 
> 
> > > I'm not sure how fruitful it would be to try to implement a pressure model given that pressure should be more or less corrected by linear advance / pressure advance.
> > 
> > 
> > If the pressure is corrected by linear advance, then there is no need for coasting.
> > Ooze is too slow and unreliable to create the end of the extrusion.
> 
> Even with a perfectly dialed linear/pressure advance coasting still comes in handy. I get my best looking seams in S3D using just a tad of coast on top of pressure advance.

This has been my experience as well. I suppose I'd rather see coasting as a feature, even if basic, rather than nothing at all. That being said a well working pressure model would be sweet! 

This was posted by a peer in a 3d printer building group. Tuned pressure advance with light coasting enabled 
![Coasting](https://user-images.githubusercontent.com/57574629/95180562-00272480-0777-11eb-9ba6-209ba93a35c2.jpg)

**Comment by Itox001** [comment]
Hi, I'm the guy that opened [issue 246](https://github.com/prusa3d/PrusaSlicer/issues/246) in PrusaSlicer. As others pointed out, it is a very useful feature so if we could finally have it here it would be awesome. I have literally waited three years for this :D

I think that a simple coast as is already found on S3D or Cura would suffice. Easier to implement and it has provided a lot of value to the other slicers. Of course @supermerill if you want to do a full-blown pressure model we will happily test and report back, but I fear that the added complexity will reflect negatively on the timeline of when we could see this feature.

**Comment by digitalentropy** [comment]
+1 on this for sure

**Comment by boromyr** [comment]
+1 for coasting

**Comment by tecnopolis-ve** [comment]
I would like to vote up for this superb feature, this one is especially useful when printing with TPU using a bowden extruder.

**Comment by wildjokerss** [comment]
just came to check on this, yep, this would be wonderful. even the simplest "coasting" would be just awesome!

**Comment by Zulaner** [comment]
I also want this feature

**Comment by supermerill** [comment]
I have to rewrite most of the gcode generator for that, so it may take a while.

**Comment by liftbag** [comment]
> I have to rewrite most of the gcode generator for that, so it may take a while.

Hopefully, I expect to be able to test it on osx when it's out, because the latest nightly builds close on opening (segmentation fault: 11)

**Comment by Xploder** [comment]
coasting would be great

**Comment by macchinette** [comment]
I too am voting in favor of the Coasting function. Thanks

**Comment by tome9111991** [comment]
im looking for an alternative for linear advance.
because have some issue with Marlin LA extruder grinding sound.. etc..

maybe kind of coasting could be usefull to reduce pressure.
wipe not very help i have pressure by change speed for exemple by overhangs.

**Comment by digitalentropy** [comment]
Re-up for coasting and whatever necessary under-the-hood changes may be needed. It can make a big difference in quality.

**Comment by supermerill** [comment]
you increase the seam-gap, it's a coasting-like thing for the external periemter.

**Comment by digitalentropy** [comment]
True, seam gap can compensate  slightly . It’s not clear if the seam gap value affects the beginning or end of the segment.

**Comment by supermerill** [comment]
end

**Comment by FDResearch** [comment]
As some feedback I have had good luck using about 220% seam gap with foaming ASA, still not perfect but this material is so weird there will always be artifacts. Honestly though on every other filament I have it off and use LA, it seems to only make gaps if your filament is properly dried.
